src/wallet/wallet.h

```
@@ -111,6 +116,13 @@ class CKeyPool
             READWRITE(nVersion);
         READWRITE(nTime);
         READWRITE(vchPubKey);
+        if (nVersion >= FEATURE_HD_SPLIT)
+            READWRITE(fInternal);
+        else
+        {
+            if (ser_action.ForRead())
+                fInternal = false;
```

Maybe throw an exception if `!ser_action.ForRead() && fInternal`?

Either way, please add braces.